Code 1

```diff
@@ -23,42 +24,43 @@ def generate_bot_response(user_input):
         return "I'm not sure how to respond to that."
 
 # Play video loop function (with video switching)
-def play_video(video_path):
+def play_video(video_path, end_event):
     cap = cv2.VideoCapture(video_path)
-    while True:
+    while not end_event.is_set():
         ret, frame = cap.read()
         if not ret:
             cap.set(cv2.CAP_PROP_POS_FRAMES, 0)
             continue
         cv2.imshow("Video", frame)
-        if cv2.waitKey(30) & 0xFF == ord('q'):
-            break
+        cv2.waitKey(30)
+    cv2.destroyAllWindows()
     cap.release()
 
 # Main program
 if __name__ == "__main__":
```

Comment:
Ech ha dini logik vo video wächsel ned ganz verstande, also hanis chli semplifiziert.

Am afang vo de loop startet idle.

Noch dem mer en Input becho händ, s'video beende und talk starte.

am schloss talk beände, idle werd am afang vo de neue loop weder gstartet